hw/ip/aes/pre_sca/alma/cpp/testbench.h

```diff
@@ -0,0 +1,73 @@
+// Copyright IAIK.
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "verilated.h"
+#include "verilated_vcd_c.h"
+
+template <class Module>
+struct Testbench {
+  unsigned long m_tickcount;
+  Module *m_core;
+  VerilatedVcdC *m_trace;
+
+  Testbench(void) {
+    Verilated::traceEverOn(true);
+    m_core = new Module;
+    m_tickcount = 0ul;
+  }
+
+  virtual ~Testbench(void) {
+    delete m_core;
+    m_core = NULL;
+  }
+
+  virtual void opentrace(const char *vcdname) {
+    if (!m_trace) {
+      m_trace = new VerilatedVcdC;
+      m_core->trace(m_trace, 99);
+      m_trace->open(vcdname);
+    }
+  }
+
+  virtual void closetrace(void) {
+    if (m_trace) {
+      m_trace->close();
+      m_trace = NULL;
```

Comment:
I think you probably want to delete `m_trace` here too: at the moment, we're leaking the `VerilatedVcdC` object.

Reply:
I think so, too. But to be honest, I am not a C++ expert at all. And I should probably have said from the beginning that I took this code from an example in the Alma repository and then adjusted it. Anyway, I've now added the `delete m_trace;` before `m_trace = NULL;`.